### PR TITLE
Trigger fix

### PR DESCRIPTION
**Pull request by sbesson**
This PR aggregates a few commits already deployed on regions CI:

- removes the Artifactory upload step, defaulting to local Maven repository
- pins the Jenkins version using the tag - upgrades require a bump PR
- fix the workflow plugin and hence the Trigger job

/cc @joshmoore @mtbc

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ slave:
         - jenkins
     volumes:
         - ./slave:/home/slave
+        - /ome:/ome
```
Comment (joshmoore) [inline docker-compose.yml]: This will likely prevent deployment on orca-3, but that I can fix locally if need be.

Comment (sbesson) [inline docker-compose.yml]: I can leave that commit out of this PR if that simplifies things.

Comment (sbesson) [inline docker-compose.yml]: Pushed away and description updated.

**Comment by joshmoore** [comment]
This does fix the `Trigger` job, but for me, it still has `OMERO-unit` in it which is no longer present.

**Comment by sbesson** [comment]
Good point. Forgot to commit the last change. Pushed now.

**Comment by joshmoore** [comment]
Merging into metadata51 (**!**) and all is working. I'll open a PR for slack next. (Perhaps I should think @manics ...)

**Pull request merged**